test/issue51.js

```diff
@@ -0,0 +1,12 @@
+var test = require('tap').test;
+var resolve = require('../');
+var path = require('path')
+
+test('issue 51', function (t) {
+    t.ok(resolve.sync('./issue51/test/') === path.resolve(__dirname, 'issue51/test/index.js'));
```

Comment:
should we also ensure that `./issue51/test/` is *not* equal to `./issue51/test` when there's a `test.js` file?